### ISSUE TITLE
Eliminar la siembra de generadores aleatorios globales

### ISSUE THREAD

**Issue by Sansanto2000** [Sansanto2000/GASP#8]
Como investigador que entrena con varios workers, quiero que el generador sea reproducible también en paralelo, para que los lotes no se corrompan al subir `workers`.

## Problema

`SpectrumLabeledSequence.__getitem__` deriva un generador aleatorio de `(seed, idx)`, pero el dibujado, el ruido y los bordes de placa siguen usando los globales `random` y `np.random`. Para atarlos al índice, `__getitem__` los siembra y restaura el estado previo al salir:

```python
estado_random = random.getstate()
estado_np = np.random.get_state()
random.seed(semilla_lote)
np.random.seed(semilla_lote)
try:
    return self._build_batch(idx, rng)
finally:
    random.setstate(estado_random)
    np.random.set_state(estado_np)
```

Eso funciona con un solo hilo, pero **no es seguro con varios**. Los globales son compartidos por todo el proceso, así que dos llamadas concurrentes entrelazan sus secuencias de guardar, sembrar y restaurar: un hilo puede guardar como "estado previo" el estado que otro hilo acaba de sembrar, y restaurarlo cuando el otro todavía lo está usando.

Medido con `ThreadPoolExecutor(max_workers=4)` sobre 8 índices, comparando contra la ejecución secuencial de los mismos índices:

| medición | resultado |
|---|---|
| intentos con al menos un lote corrupto | **6 / 6** |
| índices que difieren del resultado secuencial | **8 / 8** |

`keras.utils.Sequence` acepta `workers` y `use_multiprocessing`. Con los valores por defecto (`workers=1`, `use_multiprocessing=False`) el generador es de un solo hilo y el problema no se manifiesta. Con `workers > 1` y `use_multiprocessing=False`, Keras usa hilos del mismo proceso y la corrupción aparece. Con `use_multiprocessing=True` cada worker es un proceso aparte, con sus propios globales, y tampoco se manifiesta.

El efecto no es solo perder reproducibilidad: los lotes dejan de ser función del índice, así que reaparece el problema que la semilla venía a resolver.

## Posible origen

La aleatoriedad del camino de dibujado nunca se parametrizó. `draw_observation()`, `spectral_function()`, `add_realistic_noise()` y `add_plate_edge()` llaman directamente a `random.*` y `np.random.*`. La siembra de globales es un puente para atarlas al índice sin modificarlas.

## Posible solución

Pasar el generador aleatorio de forma explícita, eliminando por completo la siembra de globales: agregar un parámetro `rng` a `draw_observation()`, `spectral_function()`, `add_realistic_noise()` y `add_plate_edge()`, y reemplazar adentro cada `random.x()` y `np.random.x()` por el método equivalente de `rng`.

Con eso `__getitem__` se reduce a crear `rng = np.random.default_rng((self.seed, idx))` y pasarlo hacia abajo. Cada llamada trabaja sobre su propio objeto, sin estado compartido, y el problema desaparece por construcción en vez de por convención.

Conviene hacerlo en un solo cambio y no función por función: mientras alguna siga usando los globales, la siembra tiene que seguir existiendo y el defecto persiste.

Dos consecuencias a tener en cuenta:

- **Cambia todas las imágenes generadas.** `np.random.Generator` usa PCG64 y los globales usan Mersenne Twister, así que las secuencias no coinciden aunque la semilla sea la misma. Los datasets viejos y nuevos dejan de ser comparables.
- Es un cambio de API en cuatro funciones públicas. Conviene que `rng` sea *keyword-only* con default `None`, resolviendo a `np.random.default_rng()` cuando no se pasa.

## Relacionado

- #4 — el camino OBB también recibe `rng` explícito (`define_observations_limits` ya lo hace), así que ambos empujan en la misma dirección.

**Comment by Sansanto2000** [comment]
## Relacionado

- #10 — aplica el mismo criterio de diseño fuera de la librería: la configuración de la corrida entra por parámetro desde el script de entrada, no como estado global leído por `gsssp`.
